playbooks/roles/ocp-lso/tasks/main.yaml

```diff
@@ -0,0 +1,236 @@
+        - name: Setup a catalogsource for installing local storage operator
+          block:
+            - name: Include the global pull-secret update role to extract podman-secret
+              include_role:
+                name: global-secret-update
+
+        - name: Include role to create ImageContentSourcePolicy and CatalogSource
+          include_role:
+            name: set-custom-catalogsource
+          vars:
+            custom_catalogsource_name: "{{ lso_catalogsource_name }}"
+            custom_catalogsource_display_name: "lso-catalog"
+            custom_catalogsource_image: "{{ lso_catalogsource_image }}"
+          when: lso_catalogsource_image != '' and lso_catalogsource_image != None
+
+    # Verify CatalogSource creation
+    - name: Verify CatalogSource creation
+      command: oc get catsrc -A
+      register: catsrc_status
+      failed_when: catsrc_status.rc != 0  # Fail if the command fails
+
+    # Create LocalStorage Operator YAML content (OperatorGroup and Subscription)
+    - name: Create operator group for LocalStorage Operator
+      kubernetes.core.k8s:
+        state: present
+        definition:
+          apiVersion: operators.coreos.com/v1
+          kind: OperatorGroup
+          metadata:
+            name: local-operator-group
+            namespace: "{{ lso_namespace }}"
+          spec:
+            targetNamespaces:
+              - "{{ lso_namespace }}"
+
+    # Create Subscription for the LocalStorage Operator
+    - name: Create LocalStorage Operator Subscription
+      kubernetes.core.k8s:
+        state: present
+        definition:
+          apiVersion: operators.coreos.com/v1alpha1
+          kind: Subscription
+          metadata:
+            name: local-storage-operator
+            namespace: "{{ lso_namespace }}"
+          spec:
+            channel: "{{ lso_channel }}"
+            installPlanApproval: Automatic
+            name: local-storage-operator
+            source: "{{ lso_catalogsource_name }}"
+            sourceNamespace: openshift-marketplace
+
+    # Check LocalStorage Operator progress
+    - name: Check LocalStorage Operator progress
+      command: oc get ip -n "{{ lso_namespace }}"
+      register: ip_status
+      failed_when: ip_status.rc != 0  # Fail if the command fails
+
+    # Check the ClusterServiceVersion (CSV) for the LocalStorage Operator
+    - name: Check LocalStorage Operator CSV
+      command: oc get csv -n "{{ lso_namespace }}"
+      register: csv_status
+      failed_when: csv_status.rc != 0  # Fail if the command fails
+      when:
+        - lso_namespace is defined  # Ensure lso_namespace is defined
+    - name: Verify if local-storage-operator is present in the CSV output
+      debug:
+        msg: "The LocalStorage Operator CSV is available."
+      when: "'local-storage-operator' in csv_status.stdout"
+
+    # Create LocalVolumeSet YAML content for local block volumes
+    - name: Create LocalVolumeSet YAML for local block volumes
```

Comment:
Can you please use `/dev/disk/by-id` for LocalVolumeSet?  
https://bugzilla.redhat.com/show_bug.cgi?id=2273369#c5

Reply:
used one of the available from the following link